### DISCUSSION TITLE
Lack of Examples for Implementing Media App in Android Auto using .NET MAUI

### DISCUSSION THREAD

**Discussion by igorsky78** [dotnet/maui#15890] [General]
I'm currently developing a media app using .NET MAUI, specifically targeting Android Auto as one of the platforms. However, I've been struggling to find any examples or documentation related to implementing a media app for Android Auto using .NET MAUI. It seems like there is a lack of resources in this area.

I've searched the official documentation, forums, and GitHub repositories but couldn't find any relevant information or examples. I know .NET MAUI is a relatively new framework, but I was expecting at least some basic guidance on integrating with Android Auto.

Could anyone point me in the right direction or provide some insights on how to approach developing a media app for Android Auto using .NET MAUI? I would greatly appreciate any code samples, tutorials, or documentation references that can help me get started.

Thanks in advance for your assistance!

**Comment by ghost** [comment]
**Random resources, you have to put it together yourself:**
Motivator. MAUI Android Auto works.

- https://github.com/jfversluis/Plugin.Maui.Audio/blob/main/docs/audio-player.md#configure-the-playback-options
- https://developer.android.com/reference/android/media/AudioAttributes
- https://developer.android.com/reference/android/media/AudioAttributes#CONTENT_TYPE_SONIFICATION
- https://github.com/felipebaltazar/Maui.Auto.Car/
- https://github.com/felipebaltazar/Maui.Auto.Car/blob/7972f71315fba6fc0ad4c7c7099627155a701b0e/Maui.Auto.Car/Platforms/Android/Handlers/CarListViewHandler.cs
- https://github.com/mikelor/Chaaos/blob/main/src/Chaaos/Platforms/Android/Auto/Screens/MainScreen.cs
- https://github.com/felipebaltazar/Maui.Auto.Car/blob/7972f71315fba6fc0ad4c7c7099627155a701b0e/Maui.Auto.Car/Platforms/Android/Services/AndroidAutoService.cs#L30
- https://developer.android.com/training/cars/apps/iot
- https://developer.android.com/training/cars#supported-app-categories
- https://developer.android.com/training/cars/apps/iot
- https://stackoverflow.com/questions/39320048/how-to-detect-if-phone-is-connected-to-android-auto
- https://github.com/mikelor/Chaaos
- https://issuetracker.google.com/issues/363903527
- https://github.com/annikensyvertsen/AutomotiveExample?tab=readme-ov-file
- https://developer.android.com/training/cars/apps/iot
- https://developer.android.com/training/cars#supported-app-categories

**Reply by ghost** [reply]
**More resources for Distraction Optimized apps**

- [https://source.android.com/docs/automotive/driver_distraction/guidelines](https://source.android.com/docs/automotive/driver_distraction/guidelines)
- [https://source.android.com/docs/automotive/hmi/car_settings/do_car_settings](https://source.android.com/docs/automotive/hmi/car_settings/do_car_settings)
- [https://blog.csdn.net/Jason_Lee155/article/details/125867758](https://blog.csdn.net/Jason_Lee155/article/details/125867758)
- [https://blog.csdn.net/chuyouyinghe/article/details/136473472](https://blog.csdn.net/chuyouyinghe/article/details/136473472)
- [https://blog.csdn.net/qq_24252589/article/details/131288439](https://blog.csdn.net/qq_24252589/article/details/131288439)
- [https://blog.csdn.net/xdf0101/article/details/147776574](https://blog.csdn.net/xdf0101/article/details/147776574)
- [https://source.android.com/docs/automotive/driver_distraction/consume?hl=ja](https://source.android.com/docs/automotive/driver_distraction/consume?hl=ja)
- [https://stackoverflow.com/questions/65668981/how-to-get-permission-for-driver-distraction-optimized](https://stackoverflow.com/questions/65668981/how-to-get-permission-for-driver-distraction-optimized)
- [https://android.googlesource.com/platform/packages/services/Car/+/master/service/src/com/android/car/pm/CarPackageManagerService.java](https://android.googlesource.com/platform/packages/services/Car/+/master/service/src/com/android/car/pm/CarPackageManagerService.java)
- [https://blog.csdn.net/xuwb123xuwb/article/details/113949374](https://blog.csdn.net/xuwb123xuwb/article/details/113949374)
- [https://www.google.com/search?q=%3Cmeta-data+android%3Aname%3D%22distractionOptimized%22](https://www.google.com/search?q=%3Cmeta-data+android%3Aname%3D%22distractionOptimized%22)
- [https://duckduckgo.com/?q=distractionOptimized+%2B+maui+%2B+android+auto&](https://duckduckgo.com/?q=distractionOptimized+%2B+maui+%2B+android+auto&)
- [https://stackoverflow.com/questions/74348004/why-was-my-android-automotive-app-category-not-approved](https://stackoverflow.com/questions/74348004/why-was-my-android-automotive-app-category-not-approved)
- [https://issuetracker.google.com/u/0/issues/213935722](https://issuetracker.google.com/u/0/issues/213935722)
- [https://issuetracker.google.com/u/0/issues/260596923](https://issuetracker.google.com/u/0/issues/260596923)
- [https://issuetracker.google.com/issues/288913316](https://issuetracker.google.com/issues/288913316)
- [https://github.com/android/car-samples/issues/6](https://github.com/android/car-samples/issues/6)
- [https://www.polestar-forum.com/threads/android-os-simulator.517/page-5](https://www.polestar-forum.com/threads/android-os-simulator.517/page-5)
- [https://www.reddit.com/r/ex30/comments/1etpx3i/experiments_on_bringing_android_auto_to_the/](https://www.reddit.com/r/ex30/comments/1etpx3i/experiments_on_bringing_android_auto_to_the/)
- [https://developer.android.com/training/cars/apps](https://developer.android.com/training/cars/apps)

**Reply by ghost** [reply]
I also found that testing Android Auto compatible app depends on what kind of Car App Type you have. [Here you can find more details about this](https://developer.android.com/training/cars/testing#unknown-sources).

If I get this right, for example[ .POI, can not be tested from Unknown sources](https://developer.android.com/training/cars/apps#supported-app-categories).

**Usefull links about this:**
- [https://developer.android.com/training/cars/testing](https://developer.android.com/training/cars/testing)
- [https://developer.android.com/training/cars/apps](https://developer.android.com/training/cars/apps)
- [https://developer.android.com/training/cars/parked/auto](https://developer.android.com/training/cars/parked/auto)
- [https://developer.android.com/training/app-links](https://developer.android.com/training/app-links)
- [https://developer.android.com/docs/quality-guidelines/car-app-quality#IT-1](https://developer.android.com/docs/quality-guidelines/car-app-quality#IT-1)

**Reply by ghost** [reply]
Here you can find [Car app quality resources, for each category](https://developer.android.com/docs/quality-guidelines/car-app-quality#PF-1).

**Reply by ghost** [reply]
[Here is an example how to detect if the App is running on Android Auto](https://developer.android.com/training/cars/parked/auto#detect-usage).

**Reply by ghost** [reply]
I recommend adding an option for sample data, which will show this data on Android Auto, to make the Android Auto review process easier.
